lib/common/common/src/defaults.rs

```diff
@@ -4,6 +4,9 @@ use lazy_static::lazy_static;
 
 use crate::cpu;
 
+/// Current Qdrant version
+pub const QDRANT_VERSION: &str = "1.7.4";
```

Comment:
Is it now another place to remember about during version upgrades?

Reply:
Yes. We cannot access the true version from code where we need it.

I do have a trick up my sleeve though: we can assert it in `src/main.rs` to make sure it matches up with the version of the main crate. How do you feel about that?

Reply:
We can make it a global static and set it from the main crate at startup.

E.g.:

```rust
// `defaults` crate
pub static QDRANT_VERSION: Mutex<String> = Mutex::new(String::new());

// `qdrant` crate
fn main() {
    *defaults::QDRANT_VERSION.lock().unwrap() =  env!("CARGO_PKG_VERSION").into();
}
```

Or, better yet, make it three `AtomicU32`s:

```rust
// `defaults` crate
pub static QDRANT_VERSION_MAJOR: AtomicU32 = AtomicU32::new(0);
pub static QDRANT_VERSION_MINOR: AtomicU32 = AtomicU32::new(0);
pub static QDRANT_VERSION_PATCH: AtomicU32 = AtomicU32::new(0);

// `qdrant` crate
fn main() {
    defaults::QDRANT_VERSION_MAJOR.store(env!("CARGO_PKG_VERSION_MAJOR").parse().unwrap());
    defaults::QDRANT_VERSION_MINOR.store(env!("CARGO_PKG_VERSION_MINOR").parse().unwrap());
    defaults::QDRANT_VERSION_PATCH.store(env!("CARGO_PKG_VERSION_PATCH").parse().unwrap());
}
```

Reply:
I don't like that at all. It suddenly makes our sub crate dependent on our main one to work properly.

I'll quickly add an assertion in `src/main.rs` to achieve the same effect.

Reply:
>  It suddenly makes our sub crate dependent on our main one to work properly.

Kinda, but not really?

Like, we do set the values from `qdrant` crate, sure. But the variables are still accessible, they are just zeroes or empty strings by default... which I don't think is critical?

E.g., `collection` (and other `lib` crates) _don't_ run by itself, on it's own. We only run then as part of `qdrant` crate. The only exception, is when we run _tests_.

But let's assume we go with `const QDRANT_VERSION: &str = "1.2.3"`. How will you be able to use it in tests? You literally can't in any meaningful way _assert_ on `QDRANT_VERSION` value (except, _maybe_, that it's not an empty string). Otherwise, every time you release, you'd have to go and modify both `QDRANT_VERSION` *and* all tests that rely on `QDRANT_VERSION` *value*.

But if we do `static QDRANT_VERSION: Mutex<String> = ...`, then if you *want* to rely on it in tests, you can just *set* the value inside the test.

And if you never _check_ `QDRANT_VERSION` value inside your tests... than what the difference? It will be an empty string, sure, but, just run your tests with an empty string, it'll probably still work.

We pretty much already do this. We do the whole "dependency inversion" thing in a bunch of places, when we define implementation in `qdrant` crate and propagate it into `collection` crate. It's kinda like this, I think?